base/sequence_clock.go

```diff
@@ -291,6 +292,23 @@ func (c *SequenceClockImpl) UpdateWithClock(updateClock SequenceClock) {
 	}
 }
 
+func (c *SequenceClockImpl) UpdateWithPartitionClocks(partitionClocks []*PartitionClock, allowRollback bool) {
+
+	for _, partitionClock := range partitionClocks {
+		if partitionClock != nil {
+			for vbNo, seq := range *partitionClock {
+				if seq > 0 {
+					if seq < c.value[vbNo] && !allowRollback {
+						panic(fmt.Sprintf("Update attempted to set clock to earlier sequence.  Vb: %d, currentSequence: %d, newSequence: %d", vbNo, c.value[vbNo], seq))
```

Comment:
I've gone back and forth on this. With allowRollback in place, the panic here is intended to help identify significant bugs with index writing.  With a panic, it's less likely to slip through unit/functional testing uncaught. 

However, I also agree that a panic doesn't do a good job of communicating to callers (or to the SG/SGA logs) what the actual problem was, and how to fix.  I'll put the TODO in there for now.